src/pages/index.js

```diff
@@ -22,29 +32,109 @@ function handleImageClick({ name, link }) {
   imageInstance.open({ name, link });
 }
 
+function renderLoading(isLoading, btn, defaultText) {
+  if (isLoading) {
+    btn.textContent = "Saving...";
+    debugger;
+  } else {
+    btn.textContent = defaultText;
+    debugger;
+  }
+}
+
+function handleDeleteClick(evt) {
+  const element = evt.currentTarget.closest(".element");
+  console.log(element.id);
+  const popupConfirmDelete = new PopupWithConfirm(
+    "#delete-modal",
+    (element) => {
+      renderLoading(true, popupConfirmDelete.confirmBtn);
+      api
+        .deleteCard(element.id)
+        .then((message) => {
+          console.log(message);
+          renderLoading(false, popupConfirmDelete.confirmBtn, "Yes");
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+      element.remove();
+    }
+  );
+  popupConfirmDelete.setEventListeners(element);
+  popupConfirmDelete.open();
+}
+
 const cardSection = new Section(
   {
-    items: initialCards,
+    items: [],
     renderer: (item, method = "prepend", elementContainer) => {
-      const cardElement = new Card(item, "#element", handleImageClick);
+      const cardElement = new Card(
+        item,
+        "#element",
+        handleImageClick,
+        handleDeleteClick
+      );
       elementContainer[method](cardElement.createCard());
     },
   },
   ".elements__container"
 );
-cardSection.renderItems(); // generate default cards
+
+api
+  .getInitialCards()
+  .then((data) => {
+    cardSection.renderItems(data);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
 // set up user profile
 const profileInfo = new UserInfo({
   nameSelector: ".profile__name",
   captionSelector: ".profile__caption",
+  avatarSelector: ".profile__picture",
+});
+
+api.getUserInfo().then((data) => {
+  profileInfo.setUserInfo({
+    name: data.name,
+    caption: data.about,
+    avatar: data.avatar,
+  });
+});
+
+const avatarPopup = new PopupWithForm("#avatar-modal", (avatar) => {
+  renderLoading(true, avatarPopup.submitBtn);
+  profileInfo.setUserAvatar(avatar);
```

Comment:
You need to update the `DOM` only in `then` blocks after a successful response from the server, so that when there is an error, it shouldn’t change